scoreboard.pl

```diff
@@ -502,6 +502,25 @@ ($)
     return $description;
 }
 
+sub write_failed_tests_by_test
+{
+    my $directory = shift;
+    my $failed_tests_by_test_ref = shift;
+    my %failed_tests_by_test = %$failed_tests_by_test_ref;
```

Comment:
Got an error:
Ambiguous use of %{shift} resolved to %shift at
	../DOCGroup/autobuild/scoreboard.pl line 508 (#1)
    (W ambiguous) You wrote something like @{foo}, which might be
    asking for the variable @foo, or it might be calling a function
    named foo, and dereferencing it as an array reference.  If you wanted
    the variable, you can just write @foo.  If you wanted to call the
    function, write @{foo()} ... or you could just not have a variable
    and a function with the same name, and save yourself a lot of trouble.
    
Global symbol "%shift" requires explicit package name at ../DOCGroup/autobuild/scoreboard.pl line 508.
BEGIN not safe after errors--compilation aborted at ../DOCGroup/autobuild/scoreboard.pl line 1800 (#2)
    (F) You've said "use strict" or "use strict vars", which indicates 
    that all variables must either be lexically scoped (using "my" or "state"), 
    declared beforehand using "our", or explicitly qualified to say 
    which package the global variable is in (using "::").
    
Uncaught exception from user code:
	Global symbol "%shift" requires explicit package name at ../DOCGroup/autobuild/scoreboard.pl line 508.
	BEGIN not safe after errors--compilation aborted at ../DOCGroup/autobuild/scoreboard.pl line 1800.

Reply:
Sorry didn't test that; just assumed it would work. Perl's crazy syntax is tripping over itself. We have to make it explicit that it's a function:

```suggestion
    my %failed_tests_by_test = %{shift ()};
```

I did test this and it works.

Reply:
Thanks.